### PR TITLE
Remove shape functions

### PR DESCRIPTION
**Pull request by symeneses**
**What is this PR about?**
It solves #6107 

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## New features
- ...

## Bugfixes
- ...

## Documentation
- Functions deleted from the API docs.

## Maintenance
- ...

### CODE REVIEW

File: pymc/distributions/multivariate.py [hunk 1]
```diff
@@ -1651,7 +1650,9 @@ def rng_fn(cls, rng, mu, rowchol, colchol, size=None):
         output_shape = size + dist_shape
 
         # Broadcasting all parameters
-        (mu,) = broadcast_dist_samples_to(to_shape=output_shape, samples=[mu], size=size)
+        shapes = [mu.shape, output_shape]
+        sp_shapes = [s[len(size) :] if size == s[: min([len(size), len(s)])] else s for s in shapes]
```
Comment (michaelosthege) [inline pymc/distributions/multivariate.py]: Did you test this by hand?
I'd feel better it this line was covered by tests, or refactored to be easier to comprehend (or both)

**Review by michaelosthege — COMMENTED** [review]
Wow, this must be a record ratio of lines deleted vs. lines added. Awesome!

File: pymc/distributions/multivariate.py (hunk 1 above)
Comment (symeneses) [reply pymc/distributions/multivariate.py]: To test that we are getting the correct broadcastable shape, I think the best option is to keep the function `broadcast_dist_samples_shape` and its tests. `shapes_broadcasting` is replaced by `np.broadcast_shapes`. PR updated.
Also, more tests may be needed, using only what I added here was not correct but all tests were passing.

File: pymc/distributions/multivariate.py [hunk 2]
```diff
@@ -1651,7 +1651,9 @@ def rng_fn(cls, rng, mu, rowchol, colchol, size=None):
         output_shape = size + dist_shape
 
         # Broadcasting all parameters
-        (mu,) = broadcast_dist_samples_to(to_shape=output_shape, samples=[mu], size=size)
+        shapes = [mu.shape, output_shape]
+        broadcastable_shape = broadcast_dist_samples_shape(shapes, size=size)
+        mu = np.broadcast_to(mu, shape=broadcastable_shape)
```
Comment (ricardoV94) [inline pymc/distributions/multivariate.py:1656]: Is `mu = np.broadcast_to(mu, shape=dist_shape)` incorrect?

**Comment by ricardoV94** [comment]
You may be able to use this utility to broadcast everything: https://github.com/pymc-devs/pytensor/blob/63f8d6e732cd19e7f8940e25e4c44231e7765f96/pytensor/tensor/random/utils.py#L24

Although I think this RV was never properly implemented as a RandomVariable. For instance if `size=None`, the size should be given by the batch dimensions of the parameters. Also `dist_shape` should use `rowcol.shape[-1] or [-2]` not `[0]`

**Review by michaelosthege — APPROVED** [review]
LGTM

**Comment by ricardoV94** [comment]
This PR may need a rebase before we can merge

**Comment by ricardoV94** [comment]
We can rebase and merge this one as is of it's working and keep cleaning it up in future PRs

**Review by twiecki — APPROVED** [review]

**Comment by ricardoV94** [comment]
Opened an issue for the MatrixNormal, which I am not very confident is (or was) working correctly. Would have to check what actual tests there are. Otherwise the changes seem sensible. 

Thanks @symeneses

**Comment by symeneses** [comment]
> Opened an issue for the MatrixNormal, #6066 which I am not very confident is (or was) working correctly. Would have to check what actual tests there are. Otherwise the changes seem sensible.
> 
> Thanks @symeneses

#6066 is a PR not an issue. I could have a look to the tests based on your previous comment and in the new issue.

**Pull request merged**